### PR TITLE
Update 2020-7247

### PR DESCRIPTION
**Pull request by sempervictus**
#12889

### CODE REVIEW

File: modules/exploits/unix/smtp/opensmtpd_mail_from_rce.rb [hunk 1]
```diff
@@ -42,11 +42,15 @@ def initialize(info = {})
     ])
 
     register_advanced_options([
+      OptBool.new('ForceFROMFilter',    [false, 'Use Qualys\' original filter', false]),
+      OptInt.new('SafetyLines',         [true, 'Number of comment lines in body', 10]),
```
Comment (wvu-r7) [inline modules/exploits/unix/smtp/opensmtpd_mail_from_rce.rb]: [](https://www.youtube.com/watch?v=AjPau5QYtYs)

**Comment by wvu-r7** [comment]
Thanks! I've added https://github.com/rapid7/metasploit-framework/pull/12889/commits/dae06ab0c9c500a32190b3790a994f3cd8695373.

**Comment by wvu-r7** [comment]
Thanks! I've added https://github.com/rapid7/metasploit-framework/pull/12889/commits/73fcda8ae17908905ae6257b48654cefb5e4f97b.

**Comment by materaj2** [comment]
I tried to created demo docker lab with alpine:3.11 and Ubuntu:18.04, Ubuntu:19.10. (reference from this info https://www.kb.cert.org/vuls/id/390745/ ) but it's not work. The exploit is failed and I got error message such as "Sender address syntax error" (exploit from local ), "Sender domain is empty" (exploit from remote). Is there anyone have Dockerfile or smtpd.conf for CVE-2020-7247? Please Advice

**Example of my docker file**
#Dockerfile
FROM alpine:3.11
RUN apk add --no-cache python3 openssl ca-certificates opensmtpd
COPY smtpd-alpine.conf /etc/smtpd/smtpd.conf
RUN mkdir -p /var/spool/smtpd
VOLUME /var/spool/smtpd
WORKDIR /var/spool/smtpd

#smtpd-alpine.conf
table aliases file:/etc/smtpd/aliases
listen on eth0
action "local" mbox alias
action "relay" relay
match from any action "local"
match for local action "local"
match from local for any action "relay"

**Error that I got**
debug: bounce warning after 4h
debug: smtp: listen on 0.0.0.0 port 25 flags 0x400 pki "" ca ""
debug: smtp: will accept at most 524283 clients
debug: init private ssl-tree
debug: queue: done loading queue into scheduler
debug: smtpd: scanning offline queue...
debug: smtpd: offline scanning done
d2dd52cae239235f smtp connected address=127.0.0.1 host=localhost
d2dd52cae239235f smtp failed-command command="MAIL FROM:<;touch /tmp/x;>" result="553 5.1.0 Sender address syntax error"
d2dd52cae239235f smtp disconnected reason=disconnect

**Pull request merged**